index.js

```diff
@@ -194,7 +196,15 @@ NYC.prototype.instrumentAllFiles = function (input, output, cb) {
     const stats = fs.lstatSync(input)
     if (stats.isDirectory()) {
       inputDir = input
-      this.walkAllFiles(input, visitor)
+
+      const filesToInstrument = this.exclude.globSync(input)
```

Comment:
`instrumentAllFiles` is getting to be a pretty large method. I'd be tempted to potentially move this logic into `lib/instrument` and split it into a couple helper methods.